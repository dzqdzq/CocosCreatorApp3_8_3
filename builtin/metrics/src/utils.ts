import { join } from 'path';

module.paths.push(join(Editor.App.path, 'node_modules'));

import * as windows from '@base/electron-windows';

import { Display, screen } from 'electron';
import { winBounds } from './interface';

const md5: (str: string) => string = require('md5');
const getmac = require('getmac');

/**
 * 获取当前客户端的 cid
 */
export async function getClientID(): Promise<string> {
    // @ts-ignore
    return new Promise((resolve) => {
        try {
            const macAddress = getmac.default();
            return resolve(md5(macAddress));
        } catch (error) {
            console.debug(error);
            const nwInfo = require('os').networkInterfaces();
            for (const category in nwInfo) {
                if (!category) {
                    continue;
                }
                const infoArr = nwInfo[category];
                for (let i = 0; i < infoArr.length; ++i) {
                    const info: {internal: string, mac: string} = infoArr[i];
                    if (!info.internal && info.mac) {
                        return resolve(md5(info.mac));
                    }
                }
            }
            return resolve(md5('00:00:00:00:00:00'));
        }
    });
}

export function formatBytes(bytes: number, decimals = 3) {
    if (bytes === 0) return '0 Bytes';
    const k = 1024;
    const dm = decimals < 0 ? 0 : decimals;
    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
    const i = Math.floor(Math.log(bytes) / Math.log(k));
    return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
}

export function valueInvalid(value: any) {
    return value === undefined || value === null;
}

export function getMainDisplay(): Display {
    const mainUUID = windows.uuids[0];
    const mainWin = windows.uuid2win[mainUUID];
    let display = null;
    if (mainWin) {
        const mainBounds = mainWin.getBounds() as winBounds;
        const displays = screen.getAllDisplays();

        // 用主窗口左上角的点判断是否在某个窗口内
        // +2 是因为左右撑满的时候，x 会小 1-2 像素，原因未知
        for (let i = 0; i < displays.length; i++) {
            const displayBounds = displays[i].bounds as winBounds;
            if (
                mainBounds.x > displayBounds.x &&
                mainBounds.y > displayBounds.y &&
                mainBounds.x < displayBounds.x + displayBounds.width &&
                mainBounds.y < displayBounds.y + displayBounds.height
            ) {
                display = displays[i];
                break;
            }
        }
        // 左上的点没有命中的话，使用右上角的点继续判断
        if (!display) {
            for (let i = 0; i < displays.length; i++) {
                const displayBounds = displays[i].bounds as winBounds;
                if (
                    mainBounds.x + mainBounds.width > displayBounds.x &&
                    mainBounds.y > displayBounds.y &&
                    mainBounds.x + mainBounds.width < displayBounds.x + displayBounds.width &&
                    mainBounds.y < displayBounds.y + displayBounds.height
                ) {
                    display = displays[i];
                    break;
                }
            }
        }
    }
    // 如果都没有命中，则拿系统默认的主窗口
    if (!display) {
        display = screen.getPrimaryDisplay();
    }
    return display;
}
