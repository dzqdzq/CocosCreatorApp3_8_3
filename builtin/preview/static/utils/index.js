const{basename,join,extname}=require("path"),{copyFileSync,ensureDir,outputFileSync}=require("fs-extra"),EventEmitter=require("events")["EventEmitter"],TEMP_PATH=join(Editor.Project.path,"temp"),DB_PROTOCOL_HEADER="db://";async function getEnginInfo(){return await Editor.Message.request("engine","query-engine-info")}function getRightUrl(e){return e.startsWith(DB_PROTOCOL_HEADER)?e.slice(DB_PROTOCOL_HEADER.length)||(console.error("unknown mount to build: "+e),null):(console.error("unknown path to build: "+e),e)}async function writScripts(){var t=await Editor.Message.request("asset-db","query-assets",{ccType:"cc.Script"}),n=join(TEMP_PATH,"/quick-scripts");ensureDir(join(n,"assets"));for(let e=0;e<t.length;e++){var r=t[e],i=getRightUrl(r.source),s=extname(i),s=basename(i,s),o=basename(r.library[".js"]).replace(r.uuid,s),s=basename(r.library[".js.map"]).replace(r.uuid,s),i=getModules(i);outputFileSync(join(n,"assets",o),i),copyFileSync(r.library[".js.map"],join(n,"assets",s))}return n}module.exports={getEnginInfo:getEnginInfo,writScripts:writScripts};