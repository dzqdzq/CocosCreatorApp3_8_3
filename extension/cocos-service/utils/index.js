"use strict";var _0x5626a3=_0x224b;function _0x224b(e,t){var r=_0x3bcc();return(_0x224b=function(e,t){return r[e-=258]})(e,t)}function _0x4106(e,t){var r=_0x151c();return(_0x4106=function(e,t){return r[e-=416]})(e,t)}!function(){for(var e=_0x224b,t=_0x3bcc();;)try{if(231418==+parseInt(e(299))*(-parseInt(e(295))/2)+parseInt(e(270))/3+parseInt(e(283))/4*(parseInt(e(294))/5)+parseInt(e(292))/6*(parseInt(e(305))/7)+parseInt(e(284))/8+parseInt(e(285))/9+-parseInt(e(275))/10)break;t.push(t.shift())}catch(e){t.push(t.shift())}}();var _0x1409ae=_0x4106;function _0x151c(){var e=_0x224b,t=[e(281),e(274),e(261),e(265),"8104pWpYLH",e(288),e(291),e(273),"./plugin",e(277),e(260),e(263),"csEvent",e(279),e(289),e(258),e(266),e(259),e(268),"openUri",e(306),"3knrIWc",e(296),e(262),"getServiceUrls","stringify",e(304),e(267),e(269),"csAnalytics",e(271),e(293),e(309),e(276),e(302),"urls",e(298),e(301),e(278),e(264),e(300)];return(_0x151c=function(){return t})()}!function(){for(var t=_0x224b,e=_0x4106,r=_0x151c();;)try{if(403083==+parseInt(e(450))+-parseInt(e(425))/2*(-parseInt(e(448))/3)+-parseInt(e(436))/4+-parseInt(e(422))/5+-parseInt(e(428))/6+-parseInt(e(447))/7*(parseInt(e(431))/8)+-parseInt(e(433))/9*(-parseInt(e(441))/10))break;r[t(286)](r[t(280)]())}catch(e){r[t(286)](r.shift())}}();var __importDefault=this&&this[_0x5626a3(287)]||function(e){return e&&e.__esModule?e:{default:e}};Object[_0x1409ae(442)](exports,_0x1409ae(416),{value:!0}),exports[_0x1409ae(456)]=exports[_0x5626a3(268)]=exports[_0x5626a3(307)]=exports[_0x5626a3(273)]=exports.csEditor=exports[_0x1409ae(418)]=exports[_0x1409ae(439)]=exports[_0x1409ae(424)]=exports[_0x1409ae(437)]=exports[_0x1409ae(427)]=void 0;const file_1=__importDefault(require(_0x1409ae(443))),http_1=__importDefault(require(_0x1409ae(432))),util_1=__importDefault(require(_0x1409ae(438))),event_1=__importDefault(require(_0x1409ae(455))),config_1=__importDefault(require(_0x1409ae(453))),editor_1=__importDefault(require(_0x5626a3(308))),logger_1=__importDefault(require(_0x5626a3(290))),plugin_1=__importDefault(require(_0x1409ae(435))),protocol_1=__importDefault(require(_0x5626a3(272))),analytics_1=__importDefault(require(_0x1409ae(429))),request_1=require(_0x1409ae(419));function _0x3bcc(){var e=["match","2710085vKgReF","1jtxioJ","substring","getRedirectUrl","default","$plugins.","./config","2983358cbnMzN","2205MFOBiK","csPlugin","./editor","csConfig","defineProperty","ccService","csHttp","./analytics","24524LQnllV","./util","1024682RZLAHx","paramEncode","./file","apiRequest","csProtocol","./event","645618PTwOer","__esModule","./protocol","csLogger","4744644wmhTPy","6523080jJKDNS","./request","1591320luuBpe","csUtil","replace","shift","csFile","closePopup","8QUllDo","268312brzyFh","3009546LcUmHJ","push","__importDefault","./http","12580kOFdJx","./logger","13707UWfsIV","6zSwfWC","openPopup","575270rsYGNz","711430DSxiho","csEditor"];return(_0x3bcc=function(){return e})()}exports[_0x1409ae(427)]=file_1[_0x1409ae(420)],exports[_0x1409ae(437)]=http_1[_0x1409ae(420)],exports[_0x1409ae(424)]=util_1[_0x1409ae(420)],exports[_0x1409ae(439)]=event_1[_0x1409ae(420)],exports[_0x1409ae(418)]=config_1[_0x1409ae(420)],exports[_0x1409ae(449)]=editor_1[_0x1409ae(420)],exports[_0x1409ae(434)]=logger_1[_0x1409ae(420)],exports[_0x5626a3(307)]=plugin_1[_0x5626a3(302)],exports[_0x5626a3(268)]=protocol_1[_0x1409ae(420)],exports[_0x1409ae(456)]=analytics_1[_0x1409ae(420)],exports.default={csFile:file_1[_0x1409ae(420)],csHttp:http_1[_0x5626a3(302)],csUtil:util_1[_0x5626a3(302)],csEvent:event_1[_0x1409ae(420)],csConfig:config_1[_0x1409ae(420)],csEditor:editor_1[_0x1409ae(420)],csLogger:logger_1[_0x1409ae(420)],csProtocol:protocol_1[_0x1409ae(420)],csAnalytics:analytics_1[_0x1409ae(420)],urls:{},pluginTr:(e,t,r)=>editor_1[_0x1409ae(420)].tr(_0x5626a3(303)+e+"."+t,r),apiRequest:async(e,t={})=>(e[_0x5626a3(297)](/^\/api/)&&(e=e[_0x1409ae(440)](/^\/api\//,"")),(0,request_1[_0x1409ae(454)])(e,t)),async getUrl(e,t={}){var r=_0x1409ae;return"{}"===JSON[r(452)](this[r(421)])&&(this[r(421)]=await protocol_1[r(420)][r(451)]()),this[r(421)][e]?this[r(421)][e]+("{}"!==JSON[r(452)](t)?"?"+exports[r(437)][r(430)](t)[r(426)](1):""):""},async openCocosUrl(e){var t=_0x1409ae;exports[_0x5626a3(278)][t(446)](await exports[t(445)][t(423)](e))},openPopup:function(e,t){var r=_0x5626a3,_=_0x1409ae;plugin_1[r(302)][_(417)](e,t)},closePopup:function(e){var t=_0x5626a3,r=_0x1409ae;plugin_1[r(420)][t(282)](e)}},global[_0x1409ae(444)]=global[_0x1409ae(444)]||this[_0x5626a3(302)];