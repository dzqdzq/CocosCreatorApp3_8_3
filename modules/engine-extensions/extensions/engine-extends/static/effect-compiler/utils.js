const defaultOption={glsl1:!0,glsl3:!0,glsl4:!0};function stripEditorSupport(e,t=defaultOption){for(const l of e.shaders){delete l.varyings,t.glsl4||delete l.glsl4,t.glsl3||delete l.glsl3,t.glsl1||delete l.glsl1;for(const o of l.defines)delete o.defines,delete o.editor}for(const r of e.techniques)for(const d of r.passes){delete d.migrations;var s=d.properties;if(s)for(const i in s)delete s[i].editor}return delete e.dependencies,e}module.exports={stripEditorSupport:stripEditorSupport};