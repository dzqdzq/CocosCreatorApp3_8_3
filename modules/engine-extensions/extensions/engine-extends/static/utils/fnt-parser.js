"use strict";var FntLoader={INFO_EXP:/info .*?(?=\/>)|info .*/gi,COMMON_EXP:/common .*?(?=\/>)|common .*/gi,PAGE_EXP:/page .*?(?=\/>)|page .*/gi,CHAR_EXP:/char .*?(?=\/>)|char .*/gi,KERNING_EXP:/kerning .*?(?=\/>)|kerning .*/gi,ITEM_EXP:/\w+=[^ \r\n]+/gi,NUM_EXP:/^\-?\d+(?:\.\d+)?$/,_parseStrToObj:function(e){var t=e.match(this.ITEM_EXP),r={};if(t)for(var a=0,n=t.length;a<n;a++){var o=t[a],i=o.indexOf("="),s=o.substring(0,i),o=o.substring(i+1);'"'===o[0]?(o=o.substring(1,o.length-1)).match(this.NUM_EXP)&&(o=parseFloat(o)):o.match(this.NUM_EXP)&&(o=parseFloat(o)),r[s]=o}return r},parseFnt:function(e){var t=this,r={},a=e.match(t.INFO_EXP);if(a){for(var a=t._parseStrToObj(a[0]),n=t._parseStrToObj(e.match(t.COMMON_EXP)[0]),a=(r.commonHeight=n.lineHeight,r.fontSize=parseInt(a.size),cc.game.renderType===cc.game.RENDER_TYPE_WEBGL&&(a=cc.configuration.getMaxTextureSize(),n.scaleW>a.width||n.scaleH>a.height)&&Editor.log("cc.LabelBMFont._parseCommonArguments(): page can't be larger than supported"),1!==n.pages&&Editor.log("cc.LabelBMFont._parseCommonArguments(): only supports 1 page"),t._parseStrToObj(e.match(t.PAGE_EXP)[0])),o=(0!==a.id&&Editor.log("cc.LabelBMFont._parseImageFileName() : file could not be found"),r.atlasName=a.file,e.match(t.CHAR_EXP)),i=r.fontDefDictionary={},s=0,c=o.length;s<c;s++){var g=t._parseStrToObj(o[s]);i[g.id]={rect:{x:g.x,y:g.y,width:g.width,height:g.height},xOffset:g.xoffset,yOffset:g.yoffset,xAdvance:g.xadvance}}var h=r.kerningDict={},E=e.match(t.KERNING_EXP);if(E)for(s=0,c=E.length;s<c;s++){var f=t._parseStrToObj(E[s]);h[f.first<<16|65535&f.second]=f.amount}}return r}};module.exports=FntLoader;