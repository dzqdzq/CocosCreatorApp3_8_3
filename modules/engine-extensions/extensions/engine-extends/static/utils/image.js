"use strict";const getImageData=async function(e){const a=document.createElement("img");a.src=e,await new Promise((e,t)=>{a.addEventListener("load",()=>{e(a)}),a.addEventListener("error",()=>{t()})});e=document.createElement("canvas"),e.width=a.width,e.height=a.height,e=e.getContext("2d");return e.drawImage(a,0,0),e.getImageData(0,0,a.width,a.height)};module.exports={getImageData:getImageData};