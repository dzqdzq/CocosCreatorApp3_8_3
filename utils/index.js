"use strict";exports.File=require("./dist/file"),exports.Path=require("./dist/path"),exports.Math=require("./dist/math"),exports.Parse=require("./dist/parse"),exports.Url=require("./dist/url"),exports.UUID=require("./dist/uuid"),exports.Process=require("./dist/progress");