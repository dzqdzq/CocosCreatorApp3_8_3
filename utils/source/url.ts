/**
 * 快捷获取文档路径
 * @param relativeUrl 
 * @param type 
 */
export function getDocUrl(relativeUrl: string, type: 'manual' | 'api' = 'manual'): string {
    if (!relativeUrl) {
        return '';
    }
    return new URL(relativeUrl, Editor.App.urls[type]).href;
}
