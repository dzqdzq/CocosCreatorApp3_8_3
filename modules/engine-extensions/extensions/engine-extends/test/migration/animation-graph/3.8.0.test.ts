import { URL, fileURLToPath } from 'url';
import {
    migrateAnimationGraph_3_8_0,
} from '../../../source/handler/assets/migrates/animation-graph/3.8.0';
import { Archive } from '../../../source/handler/assets/utils/migration-utils';
import fs from 'fs-extra';
import ps from 'path';

test('Migrate animation graph 3.8.0', async () => {
    const data = await fs.readJson(ps.join(
        fileURLToPath(new URL('../test-inputs', import.meta.url)),
        'target-3.8.0',
        'source-3.7.2',
        'assets',
        'animation-graphs',
        'default.animgraph',
    ));
    const archive = new Archive(data);
    migrateAnimationGraph_3_8_0(archive);
    const output = JSON.parse(JSON.stringify(archive.get()));
    expect(output).toMatchSnapshot();
});
