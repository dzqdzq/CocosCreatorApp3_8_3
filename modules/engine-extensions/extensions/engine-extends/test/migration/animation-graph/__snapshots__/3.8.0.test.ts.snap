// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Migrate animation graph 3.8.0 1`] = `
[
  {
    "__type__": "cc.animation.AnimationGraph",
    "_layers": [
      {
        "__id__": 1,
      },
    ],
    "_name": "",
    "_native": "",
    "_objFlags": 0,
    "_variables": {
      "v_b": {
        "__id__": 22,
      },
      "v_f": {
        "__id__": 20,
      },
      "v_i": {
        "__id__": 21,
      },
    },
  },
  {
    "__type__": "cc.animation.Layer",
    "_stateMachine": {
      "__id__": 2,
    },
    "mask": null,
    "name": "",
    "weight": 1,
  },
  {
    "__editorExtras__": {
      "clone": null,
      "id": "16823108853190.22237115162801335",
      "name": "",
      "viewport": {
        "left": -122,
        "scale": 1,
        "top": 10,
      },
    },
    "__type__": "cc.animation.StateMachine",
    "_anyState": {
      "__id__": 5,
    },
    "_entryState": {
      "__id__": 3,
    },
    "_exitState": {
      "__id__": 4,
    },
    "_states": [
      {
        "__id__": 3,
      },
      {
        "__id__": 4,
      },
      {
        "__id__": 5,
      },
      {
        "__id__": 6,
      },
      {
        "__id__": 7,
      },
      {
        "__id__": 8,
      },
    ],
    "_transitions": [
      {
        "__id__": 9,
      },
      {
        "__id__": 14,
      },
      {
        "__id__": 17,
      },
    ],
  },
  {
    "__editorExtras__": {
      "centerX": -125,
      "centerY": 0,
      "clone": null,
      "id": "16823108853190.0030910070942671286",
      "name": "",
    },
    "__type__": "cc.animation.State",
    "name": "Entry",
  },
  {
    "__editorExtras__": {
      "centerX": 125,
      "centerY": 0,
      "clone": null,
      "id": "16823108853190.9319504307124125",
      "name": "",
    },
    "__type__": "cc.animation.State",
    "name": "Exit",
  },
  {
    "__editorExtras__": {
      "centerX": 419,
      "centerY": 40,
      "clone": null,
      "id": "16823108853190.9680302262104088",
      "name": "",
    },
    "__type__": "cc.animation.State",
    "name": "Any",
  },
  {
    "__editorExtras__": {
      "centerX": 81.5,
      "centerY": -115,
      "clone": null,
      "id": "16823109123240.07731438174679583",
      "name": "valid_conditions",
    },
    "__type__": "cc.animation.EmptyState",
    "name": "valid_conditions",
  },
  {
    "__editorExtras__": {
      "centerX": 135.5,
      "centerY": -13,
      "clone": null,
      "id": "16823111478290.9709105624785797",
      "name": "invalid_condition_unexisting_variable",
    },
    "__type__": "cc.animation.EmptyState",
    "name": "invalid_condition_unexisting_variable",
  },
  {
    "__editorExtras__": {
      "centerX": 94.5,
      "centerY": 122,
      "clone": null,
      "id": "16823112500760.7230973552240467",
      "name": "invalid_condition_mismatched_type",
    },
    "__type__": "cc.animation.EmptyState",
    "name": "invalid_condition_mismatched_type",
  },
  {
    "__editorExtras__": {
      "clone": null,
      "id": "16823111478290.014899838991285996",
      "name": "",
    },
    "__type__": "cc.animation.Transition",
    "conditions": [
      {
        "__id__": 10,
      },
      {
        "__id__": 12,
      },
    ],
    "from": {
      "__id__": 3,
    },
    "to": {
      "__id__": 6,
    },
  },
  {
    "__type__": "cc.animation.BinaryCondition",
    "lhs": 0,
    "lhsBinding": {
      "__id__": 11,
    },
    "operator": 1,
    "rhs": 1,
  },
  {
    "__type__": "cc.animation.TCVariableBinding",
    "type": 0,
    "variableName": "v_f",
  },
  {
    "__type__": "cc.animation.BinaryCondition",
    "lhs": 0,
    "lhsBinding": {
      "__id__": 13,
    },
    "operator": 0,
    "rhs": 2,
  },
  {
    "__type__": "cc.animation.TCVariableBinding",
    "type": 3,
    "variableName": "v_i",
  },
  {
    "__editorExtras__": null,
    "__type__": "cc.animation.Transition",
    "conditions": [
      {
        "__id__": 15,
      },
    ],
    "from": {
      "__id__": 3,
    },
    "to": {
      "__id__": 7,
    },
  },
  {
    "__type__": "cc.animation.BinaryCondition",
    "lhs": 0,
    "lhsBinding": {
      "__id__": 16,
    },
    "operator": 3,
    "rhs": 3,
  },
  {
    "__type__": "cc.animation.TCVariableBinding",
    "type": 0,
    "variableName": "v_unexisting",
  },
  {
    "__editorExtras__": null,
    "__type__": "cc.animation.Transition",
    "conditions": [
      {
        "__id__": 18,
      },
    ],
    "from": {
      "__id__": 3,
    },
    "to": {
      "__id__": 8,
    },
  },
  {
    "__type__": "cc.animation.BinaryCondition",
    "lhs": 0,
    "lhsBinding": {
      "__id__": 19,
    },
    "operator": 2,
    "rhs": 4,
  },
  {
    "__type__": "cc.animation.TCVariableBinding",
    "type": 0,
    "variableName": "v_b",
  },
  {
    "__type__": "cc.animation.PlainVariable",
    "_type": 0,
    "_value": 0,
  },
  {
    "__type__": "cc.animation.PlainVariable",
    "_type": 3,
    "_value": 0,
  },
  {
    "__type__": "cc.animation.PlainVariable",
    "_type": 1,
    "_value": false,
  },
]
`;
